main.tf

```diff
@@ -143,36 +143,55 @@ data "aws_ami" "user_ami" {
   }
 }
 
-resource "aws_launch_configuration" "container_instance" {
-  lifecycle {
-    create_before_destroy = true
+resource "aws_launch_template" "container_instance" {
+  block_device_mappings {
+    device_name = "/dev/sda1"
+
+    ebs {
+      volume_type = "${var.root_block_device_type}"
+      volume_size = "${var.root_block_device_size}"
+    }
   }
 
-  root_block_device {
-    volume_type = "${var.root_block_device_type}"
-    volume_size = "${var.root_block_device_size}"
+  credit_specification {
+    cpu_credits = "${var.unlimited_cpu ? "unlimited" : "standard"}"
   }
 
-  name_prefix          = "lc${title(var.environment)}ContainerInstance-"
-  iam_instance_profile = "${aws_iam_instance_profile.container_instance.name}"
+  disable_api_termination = false
```

Comment:
I'm not sure if we protected EC2 instances from being terminated via the API with launch_template.

Reply:
This seems like a good decision for now. In the future, we can consider exposing it as a top-level variable, but for now leaving it as `false` makes sense.